client/jetpack-cloud/sections/partner-portal/lib/querystring-products.ts

```diff
@@ -0,0 +1,22 @@
+import { type IssueLicenseRequest } from '../hooks/use-issue-licenses';
+
+export const serializeQueryStringProducts = ( products: IssueLicenseRequest[] ): string => {
+	return products.map( ( { slug, quantity } ) => `${ slug }:${ quantity }` ).join( ',' );
+};
+
+export const parseQueryStringProducts = (
+	products: string | null | undefined
+): IssueLicenseRequest[] => {
+	if ( ! products ) {
+		return [];
+	}
+
+	const commaSeparated = products.split( ',' );
+	return commaSeparated.map( ( i ) => {
+		const colonSeparatedItem = i.split( ':' );
+		const quantity = parseInt( colonSeparatedItem[ 1 ] ?? 1, 10 );
+		const slug = colonSeparatedItem[ 0 ];
+
+		return { slug, quantity };
```

Comment:
How about we rewrite this, including the error scenario of `parseInt` here? WDYT?

```suggestion
		const [slug, quantityString] = item.split(':');
		let quantity = parseInt(quantityString, 10);
		
		 if (isNaN(quantity)) {
            quantity = 1;
         }
         
		return { slug, quantity };
```